game/src/graphic/textures/TextureHandler.java

```diff
@@ -15,17 +18,44 @@
 /**
  * A texture handler class for managing textures paths for further use.
  *
- * <p>Singleton (possibly thread-safe).
+ * <p>Singleton.
  */
 public class TextureHandler {
+    public static final String PLACEHOLDER_FILENAME = ".resource_root_please_not_modify";
     private static final TextureHandler INSTANCE = new TextureHandler();
 
     private final Map<String, Set<FileHandle>> pathMap = new LinkedHashMap<>();
 
     private TextureHandler() {
-        addAllAssets(Gdx.files.internal(Gdx.files.getLocalStoragePath()));
+        List<Path> placeholderPaths = findAllPlaceholderPaths();
+        assert placeholderPaths.size() == 2;
+        // takes the placeholder path with the longest path string...
+        if (placeholderPaths.get(0).toString().length()
+                > placeholderPaths.get(1).toString().length()) {
+            addAllAssets(new FileHandle(placeholderPaths.get(0).getParent().toFile()));
+        } else {
+            addAllAssets(new FileHandle(placeholderPaths.get(1).getParent().toFile()));
+        }
+    }
+
+    private List<Path> findAllPlaceholderPaths() {
+        try (Stream<Path> walk = Files.walk(Path.of(Gdx.files.getLocalStoragePath()), 4)) {
```

Comment:
warum "4"? das gehört mindestens dokumentiert!

Reply:
habe keine ide offen. was ist der rückgabetyp der gdx-methode? vom namen her müsste hier doch bereits ein path kommen, so dass das path.of() redundant ist?!

Reply:
Gdx.files.getLocalStoragePath() gibt einen String zurück, Path.of( ) erwartet einen String.

Beispielsweise würde

    System.out.println(Path.of(Gdx.files.getLocalStoragePath()));
    System.out.println(new File("").getAbsolutePath());

die gleiche Ausgabe erzeugen, bei mir: `C:\Users\tobi\path\to\Dungeon`

vom content root aus werden noch 4 Schritte bis nach `Dungeon\build\resources\main\.resource_root_please_not_modify` gebraucht

das sollte logisch so stimmen

Reply:
libgdx ist echt schrott. bei "getPath" erwartet man in java einen path und keinen string. aber dann passt das ja. wollte nur sicherheitshalber nachfragen, weil der rechner aus ist.

Reply:
das mit der 4 ist hässlich. das ist eine konfiguration, und hier wird mitten im code ein magic number reingecoded. das geht so nicht! sobald sich an der konfiguration was ändert (weil vielleicht das projektlayout anders sein soll), knallt dir hier der code komplett unerwartet weg.

- dazu gehört mind. ein ausführlicher javadoc.
- projektlayout ist in der build.gradle definiert. hier sollte der pfad spezifiziert werden und dann im code genutzt werden.

Reply:
@cagix 

> projektlayout ist in der build.gradle definiert. hier sollte der pfad spezifiziert werden und dann im code genutzt werden.

Das war ja mitunter das Problem. Deshalb ja auch dieser PR. Das build-Verzeichnis bzw. das asset-Verzeichnis kann in der build.gradle festgelegt werden. Aber an gradle-Variablen kommt man -- **mWn** -- nur innerhalb der Java-Anwendung heran, wenn die Variablen vorher als command line arguments an die main-Methode weitergegeben wurden. Möchten wir das? Dann würden die Tests auch net mehr funktionieren.

Vereinfacht gesagt, sollte die Java-Anwendung vielleicht ihre ausführende Umgebung nicht kennen.

Reply:
genau, die anwendung sollte die umgebung nicht kennen.

aber jetzt deshalb annahmen tief als magic im code zu verbuddeln und diese nicht annähernd zu dokumentieren ist auch keine lösung.

vielleicht ist das der punkt, wo wir (a) über parameter für den starter nachdenken sollten oder (b) den pfad in game bei der initialisierung/konfiguration setzen sollten. beides ist immer noch besser als der aktuelle ansatz. @AMatutat als "architekt", jetzt bist du wirklich gefragt: a oder b? oder c (und wie sieht das aus?)

Reply:
Hab das mal in https://github.com/Programmiermethoden/Dungeon/issues/587 gepackt.
Hier ist soviel Diskutiert worden, das GitHub garnicht mehr klar kommt